lib/connectors/memory.js

```diff
@@ -250,6 +250,46 @@ Memory.prototype.updateOrCreate = function(model, data, options, callback) {
     }
   });
 };
+Memory.prototype.patchOrCreateWithWhere =
+  Memory.prototype.upsertWithWhere = function(model, where, data, options, callback) {
+    var self = this;
+    var element = Object.keys(where)[0];
```

Comment:
Still I don't agree with this. 
We support `AND` and `OR` conditions in `where` clause; Example:

```
    where: {
      or: {
        'name': {like: 'foo'},
        'familyName': {like: 'bar'}
      }
    }
```

Needs @bajtos thought.